version: "3.9"

services:
  db:
    image: postgres
    ports:
      - "5432:5432"
    volumes:
      - ./data/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB
      - POSTGRES_USER
      - POSTGRES_PASSWORD
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  server:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
    ports:
      - 8000:8000
    environment:
      - PREVIEW_URL
      - DEPLOY_HOOK
      - DEBUG
      - SECRET_KEY
      - POSTGRES_HOST
      - POSTGRES_PORT
      - POSTGRES_NAME
      - POSTGRES_USER
      - POSTGRES_PASSWORD
      - AWS_S3_REGION_NAME
      - AWS_S3_ENDPOINT_URL
      - AWS_STORAGE_BUCKET_NAME
      - AWS_ACCESS_KEY_ID
      - AWS_SECRET_ACCESS_KEY
    depends_on:
      db:
        condition: service_healthy
    